startup.py

```diff
@@ -0,0 +1,127 @@
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class PhotoshopLauncher(SoftwareLauncher):
+    """
+    Handles the launching of Photoshop. Contains the logic for
+    scanning for installed versions of the software and
+    how to correctly set up a launch environment for the tk-photoshopcc
+    engine.
+    """
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        icon_path = os.path.join(self.disk_location, "resources", "ps_2017_icon_256.png")
+
+        if sys.platform == "darwin":
+            # Default installs are located here:
+            # /Applications/Adobe Photoshop CC 2017/Adobe Photoshop CC 2017.app
+            glob_pattern = "/Applications/Adobe Photoshop CC [0-9][0-9][0-9][0-9]/Adobe Photoshop CC [0-9][0-9][0-9][0-9].app"
+            version_regex = re.compile(
+                "^/Applications/Adobe Photoshop CC ([0-9]{4})/Adobe Photoshop CC ([0-9]{4})\.app$",
+                re.IGNORECASE
+            )
+
+        elif sys.platform == "win32":
+            # Default installs are located here:
+            # C:\program files\Adobe\Adobe Photoshop CC 2017\Photoshop.exe
+            glob_pattern = "C:\\Program Files\\Adobe\\Adobe Photoshop CC [0-9][0-9][0-9][0-9]\\Photoshop.exe"
+            version_regex = re.compile(
+                "^C:\\\\Program Files\\\\Adobe\\\\Adobe Photoshop CC ([0-9]{4})\\\\Photoshop.exe$",
+                re.IGNORECASE
+            )
+
+        else:
+            self.logger.debug("Photoshop not supported on this platform.")
+            return []
+
+        self.logger.debug("Scanning for photoshop installations in '%s'..." % glob_pattern)
+        paths = glob.glob(glob_pattern)
+
+        software_versions = []
+        for path in paths:
+            # extract version number
+            self.logger.debug("Found photoshop install in '%s'" % path)
+
+            match = version_regex.match(path)
+            if match:
+                # extract first group to get version string
+                dcc_version = match.group(1)
+                self.logger.debug("This is version '%s'" % dcc_version)
+
+                # see if we have a version filter
+                if versions and dcc_version in versions:
```

Comment:
Shouldn't this be `not in`? i.e. version constraint was supplied and this matched version *is not* in the list?